src/vm/assemblyspec.cpp

```diff
@@ -1560,7 +1559,12 @@ BOOL AssemblySpecBindingCache::StoreAssembly(AssemblySpec *pSpec, DomainAssembly
     UPTR key = (UPTR)pSpec->Hash();
 
     // On CoreCLR, we will use the BinderID as the key 
-    ICLRPrivBinder* pBinderContextForLookup = pAssembly->GetFile()->GetBindingContext();
+    ICLRPrivBinder* pBinderContextForLookup = pSpec->GetBindingContext();
```

Comment:
Why can't we just use `pAssembly->GetFile()->GetBindingContext()` as before?

Reply:
It doesn't seem unreasonable that there may be a bug here, but I don't think its related to unloadability. This change is plenty complicated without adding possibly unrelated changes to it. @janvorli could you look into if this is necessary for the scenarios which you are testing?

Reply:
Reverting this change doesn't have any impact on my testing scenarios.